src/components/navBar/navBar.js

```diff
@@ -0,0 +1,400 @@
+      if (oldTab) {
+        oldTab.setSelected(false);
+        oldIndex = tabs.indexOf(oldTab);
+      }
+    }
+
+    if (newValue) {
+      var tab = getTabByName_(newValue);
+      if (tab) {
+        tab.setSelected(true);
+        var newIndex = tabs.indexOf(tab);
+        ctrl.timeout_(function() {
+          updateInkBarStyles(tab, newIndex, oldIndex);
+        });
+      }
+    }
+  }
+
+  /**
+   * Repositions the ink bar to the selected tab.
+   * "borrowed" from ng-material
+   * @export
+   */
+  function updateInkBarStyles(tab, newIndex, oldIndex) {
+    var tabEl = tab.getButtonEl();
+    var left = tabEl.offsetLeft;
+
+    ctrl.inkbar_.toggleClass('md-left', newIndex < oldIndex)
+        .toggleClass('md-right', newIndex > oldIndex);
+    ctrl.inkbar_.css({left: left + 'px', width: tabEl.offsetWidth + 'px'});
+  }
+
+  /**
+   * Returns an array of the current tabs.
+   * @return {!Array<!NavLinkController>}
+   * @private
+   */
+  function getTabs_() {
+    var linkArray = Array.prototype.slice.call(
+        ctrl.element_.querySelectorAll('.md-nav-link'));
+    return linkArray.map(function(el) {
+      return angular.element(el).controller('mdNavLink')
+    });
+  }
+
+  /**
+   * Returns the tab with the specified name.
+   * @param {string} name The name of the tab, found in its data-name attribute.
+   * @return {!NavLinkController|undefined}
+   * @private
+   */
+  function getTabByName_(name) {
+    return findTab_(function(tab) { return tab.getName() == name; });
+  }
+
+  /**
+   * Returns the selected tab.
+   * @return {!NavLinkController|undefined}
+   */
+  function getSelectedTab_() {
+    return findTab_(function(tab) { return tab.isSelected() });
+  }
+
+  /**
+   * Returns the focused tab.
+   * @return {!NavLinkController|undefined}
+   * @export
+   */
+  function getFocusedTab_() {
+    return findTab_(function(tab) { return tab.hasFocus() });
+  }
+
+  function findTab_(fn) {
+    var tabs = getTabs_();
+    for (var i = 0; i < tabs.length; i++) {
+      if (fn(tabs[i])) {
+        return tabs[i];
+      }
+    }
+
+    return null;
+  }
+
+  /**
+   * Direct focus to the selected tab when focus enters the nav bar.
+   * @export
+   */
+  function onFocus() {
+    var tab = getSelectedTab_();
+    if (tab) {
+      tab.setFocused(true);
+    }
+  }
+
+  /**
+   * Clear tab focus when focus leaves the nav bar.
+   * @export
+   */
+  function onBlur() {
+    var tab = getFocusedTab_();
+    if (tab) {
+      tab.setFocused(false);
+    }
+  }
+
+  /**
+   * Move focus from oldTab to newTab.
+   * @param {!NavLinkController} oldTab
+   * @param {!NavLinkController} newTab
+   */
+  function moveFocus(oldTab, newTab) {
+    oldTab.setFocused(false);
+    newTab.setFocused(true);
+  }
+
+  /**
+   * Responds to keypress events.
+   * @param {!jQuery.Event} e
+   * @export
+   */
+  function onKeydown(e) {
+    var tabs = getTabs_();
+    var focusedTab = googAsserts.assert(getFocusedTab_());
```

Comment:
Where is this assert function coming from? Wherever it is, it shouldn't be called `googAsserts`